### PR TITLE
seqan 2.4.0(new formula)

### PR DESCRIPTION
**Pull request by rrahn**
- [x] Have you followed the [guidelines for contributing](https://github.com/brewsci/homebrew-bio/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/brewsci/homebrew-bio/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source FORMULA`, where `FORMULA` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict FORMULA` (after doing `brew install FORMULA`)?

### Note
~Currently brew audit fails: the related issue is documented here: https://github.com/Homebrew/brew/issues/5561; all suggested solutions did not work for me, but I copied the formula from the science repo and made small adaptions. Any suggestions on getting audit to work is very appreciated.~

-----

### CODE REVIEW

File: Formula/seqan.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+class Seqan < Formula
+  desc "C++ library of sequence algorithms and data structures"
+  homepage "https://www.seqan.de/"
+  # doi "10.1186/1471-2105-9-11"
+  # tag "bioinformatics"
```
Comment (sjackman) [inline Formula/seqan.rb]: Please delete this line.

File: Formula/seqan.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  # doi "https://doi.org/10.1016/j.jbiotec.2017.07.017"
+  # tag "Journal of Biotechnology"
```
Comment (sjackman) [inline Formula/seqan.rb]: Please delete this line.

File: Formula/seqan.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
+class Seqan < Formula
+  desc "C++ library of sequence algorithms and data structures"
+  homepage "https://www.seqan.de/"
+  # doi "10.1186/1471-2105-9-11"
```
Comment (sjackman) [inline Formula/seqan.rb]: Please use `brew tap brewsci/base` and then…
```console
$ brew cite --ruby 10.1186/1471-2105-9-11
# cite D_ring_2008: "https://doi.org/10.1186/1471-2105-9-11"
```
```suggestion
  # cite D_ring_2008: "https://doi.org/10.1186/1471-2105-9-11"
```

File: Formula/seqan.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  # tag "bioinformatics"
+  # doi "https://doi.org/10.1016/j.jbiotec.2017.07.017"
```
Comment (sjackman) [inline Formula/seqan.rb]: Same as above use `brew cite --ruby 10.1016/j.jbiotec.2017.07.017`

Comment (sjackman) [reply Formula/seqan.rb]: Please move these `# cite` comments below `head`.

File: Formula/seqan.rb [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  # tag "Journal of Biotechnology"
+
```
Comment (sjackman) [inline Formula/seqan.rb]: Please delete this blank line.

File: Formula/seqan.rb [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  url "https://github.com/seqan/seqan/releases/download/seqan-v2.4.0/seqan-library-2.4.0.tar.xz"
+  sha256 "dd97b1514ab83acb7d7be911b157979e188e8ca72cc61c430c1e0fd03bcd41a5"
+  head "https://github.com/seqan/seqan.git"
+
+  # seqan-library installs only header files.
+  bottle :unneeded
```
Comment (sjackman) [inline Formula/seqan.rb]: Please remove these three lines 12-14.

File: Formula/seqan.rb [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    else
+      prefix.install_metafiles "share/doc/seqan"
+      doc.install Dir["share/doc/seqan/*"]
+      lib.install "share/pkgconfig", "share/cmake"
+    end
+  end
+
+  test do
+    # seqan-library installs only header files, so we test if version.h exists and has the right version number.
+    system "test -e filename", include/"seqan/version.h"
```
Comment (sjackman) [inline Formula/seqan.rb]:
```suggestion
    assert_predicate include/"seqan/version.h", :exist?
```

File: Formula/seqan.rb [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+    system "grep \"SEQAN_VERSION_MAJOR 2\"", include/"seqan/version.h"
```
Comment (sjackman) [inline Formula/seqan.rb]: Please do not include the version number in the test.
```suggestion
    assert_match "SEQAN_VERSION_MAJOR", include/"seqan/version.h"
```

File: Formula/seqan.rb [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+    system "grep \"SEQAN_VERSION_MINOR 4\"", include/"seqan/version.h"
+    system "grep \"SEQAN_VERSION_PATCH 0\"", include/"seqan/version.h"
```
Comment (sjackman) [inline Formula/seqan.rb]: Please delete these two lines 32 and 33.

**Comment by sjackman** [comment]
> Currently brew audit fails: the related issue is documented here: [Homebrew/brew#5561](https://github.com/Homebrew/brew/issues/5561); all suggested solutions did not work for me, but I copied the formula from the science repo and made small adaptions. Any suggestions on getting audit to work is very appreciated.

Try…
```sh
rm -rf /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby
rm -rf /usr/local/Homebrew/Library/Homebrew/vendor/bundle
brew style hello
```

**Comment by rrahn** [comment]
@sjackman I did a complete fresh install of homebrew and I could get audit to work this way. I also adapted the test as you suggested, though I needed to read in the version file, otherwise the assertion failed as the pattern was matched against the path name and not the file content. Please let me know, if this needs to be revised some how. I am very new to ruby 😅. And many thanks for the quick response and review. 👍

**Comment by sjackman** [comment]
Merged. Thank you for contribution to Brewsci/bio, @rrahn ! 🏆

**Comment by rrahn** [comment]
@sjackman you are welcome and many thanks for the quick support.

**Pull request closed without merging**